### DISCUSSION TITLE
Speed of the download and processing to slow, what is the solution?

### DISCUSSION THREAD

**Discussion by dzungpv** [open-meteo/open-meteo#379] [Q&A] [answered]
I tested the code on a Ubuntu server in Singapore, the download speed very slow, it take 24 hours and not complete the dataset.
So now I only want running forecast API like here https://open-meteo.com/en/docs with full "Daily Weather Variables" and "Hourly Weather Variables".
I read the docs with the parameter 
```
--only-variables  variable1,variable2
```
we can limit the size of the download data, so what the variable each data-source I need to download?

I don't know what exactly what variables from each data-source to produce all the data in forecast API.

**Comment by patrick-zippenfenig** [comment, marked as answer]
This download speed is expected for a remote server. I rent different servers as close as possible to the originating datacenters from the national weather services (Europe and North America).

Per default, only surface weather variables are downloaded and not all model and pressure levels (roughly only 20% of the total dataset). To get all data, you will even have to download way more data.

One possible solution would be an AWS open data hosting (see #101) to distribute the converted database. This is however not foreseeable in the near future.

**Reply by dzungpv** [reply]
Thank you for clear answer.
